src/client/utils/url-resolver.js

```diff
@@ -0,0 +1,62 @@
+import nativeMethods from '../sandbox/native-methods';
+import { ensureTrailingSlash } from '../../utils/url';
+
+const DOCUMENT_URL_RESOLVER = 'hammerhead|document-url-resolver';
+
+var urlResolver = {
```

Comment:
No. This was a bad practice. Since it's a singletone it can be just plain object. No need for class here.